Sources/AvatarView.swift

```diff
@@ -25,33 +25,72 @@
 import Foundation
 
 open class AvatarView: UIView {
-
+    
     // MARK: - Properties
-
+    
+    public var adjustsFontSizeToFitWidth = true
     open var avatar: Avatar = Avatar()
     open var imageView = UIImageView()
+    private var radius: CGFloat?
+    private var font: UIFont? {
+        willSet {
+            if newValue != nil {
+                print("Initialized Font Size: \(String(describing: newValue?.pointSize))")
+                initialFontSize = (newValue?.pointSize)!
+            }
+        }
+    }
+    
+    open var minimumScaleFactor: CGFloat = 1.0
+    private var initialFontSize: CGFloat? {
+        willSet {
+            if newValue != nil {
+                minimumFontSize = newValue! * minimumScaleFactor
+            }
+        }
+    }
+    
+    private var minimumFontSize: CGFloat?
+    // MARK: - Overrided Properties
+    open override var frame: CGRect {
+        didSet {
+            imageView.frame = bounds
+            setCorner(radius: self.radius)
+        }
+    }
     
     // MARK: - Initializers
-    override init(frame: CGRect) {
+    override public init(frame: CGRect) {
         super.init(frame: frame)
-        prepareView()
+//        prepareView()
     }
     
     convenience public init() {
-        let frame = CGRect(x: 0, y: 0, width: 30, height: 30)
-        self.init(frame: frame)
-        prepareView()
+        self.init(frame: .zero)
     }
     
     private func getImageFrom(initals: String, withColor color: UIColor = UIColor.white, fontSize: CGFloat = 14) -> UIImage {
-        _ = UIGraphicsBeginImageContext(CGSize(width: 30, height: 30))
+        let width = frame.width
+        let height = frame.height
+        if width == 0 || height == 0 {return UIImage()}
+        if font == nil {
+            initialFontSize = fontSize
+            font = UIFont.systemFont(ofSize: fontSize)
+        }
+        
+        _ = UIGraphicsBeginImageContext(CGSize(width: width, height: height))
         let context = UIGraphicsGetCurrentContext()!
         
         //// Text Drawing
-        let textRect = CGRect(x: 5, y: 6, width: 20, height: 20)
+        let textRect = calculateTextRect(outerViewWidth: width, outerViewHeight: height)
+        if adjustsFontSizeToFitWidth {
+            let newFontSize = calculateFontSize(text: initals, font: font!, width: textRect.width, height: textRect.height)
```

Comment:
I would love to avoid `force unwrapping optionals` I prefer `guard` statements for safety. ⛑